### PR TITLE
add more registers & refactor splitter

### PR DESCRIPTION
**Pull request by rayrayraykk**
# Main changes:
1. Add more registers;
2. Refactor the splitters with an ABC BaseSplitter.
3. remove `graph_type` splitter, which is a duplicate of `LDA` splitter.

### CODE REVIEW

**Review by xieyxclack — COMMENTED** [review]
The new registers look good to me, good job!

**Comment by rayrayraykk** [comment]
I have updated this PR according to our discussion @joneswong.
1. remove `kwargs` in `__init__`.
2. remove `graph_type` splitter, which is a duplicate of `LDA` splitter.

**Review by xieyxclack — APPROVED** [review]
LGTM

File: federatedscope/core/splitters/graph/randchunk_splitter.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import numpy as np
 
+from torch_geometric.transforms import BaseTransform
 
-class RandChunkSplitter:
+
+class RandChunkSplitter(BaseTransform):
```
Comment (joneswong) [inline federatedscope/core/splitters/graph/randchunk_splitter.py:6]: this class has no need to be a subclass of `BaseSplitter`?

**Review by joneswong — APPROVED** [review]
LGTM

**Pull request merged**